configure.ac

```diff
@@ -4943,6 +4943,21 @@ PY_CHECK_FUNC([setgroups], [
 #endif
 ])
 
+# check for posix_openpt, grantpt, unlockpt, and ptsname
+
+AC_SEARCH_LIBS([posix_openpt], [util],
+ [AC_DEFINE([HAVE_POSIX_OPENPT], [1], [Define to 1 if you have the `posix_openpt' function.])]
+)
+AC_SEARCH_LIBS([grantpt], [util],
+ [AC_DEFINE([HAVE_GRANTPT], [1], [Define to 1 if you have the `grantpt' function.])]
+)
+AC_SEARCH_LIBS([unlockpt], [util],
+ [AC_DEFINE([HAVE_UNLOCKPT], [1], [Define to 1 if you have the `unlockpt' function.])]
+)
+AC_SEARCH_LIBS([ptsname], [util],
+ [AC_DEFINE([HAVE_PTSNAME], [1], [Define to 1 if you have the `ptsname' function.])]
+)
```

Comment:
IIRC, AC_SEARCH_LIBS affects the global LIBS. We probably do not want that.

Reply:
Thank you for the review. I has added `os.login_tty()` in this PR: https://github.com/python/cpython/pull/29658, where I had originally used `AC_CHECK_FUNCS`. However, @tiran suggested that I use `AC_SEARCH_LIBS` instead, mentioning the following:

> This approach is easier and better. It updates LIBS only when it is necessary and won't add libutils a second time. The login_tty is provided by libutil on Linux, FreeBSD, OpenBSD, and NetBSD.

Should I still go ahead and change them to `AC_CHECK_FUNCS`? Or is there yet another approach that is better?

Reply:
You could wrap this whole section with `WITH_SAVE_ENV` just to be sure LIBS (or anything else) is touched. You can grep `configure.ac` for example use.

Reply:
Thank you for your guidance. I will do this now.